src/responder/pam/pamsrv_json.c

```diff
@@ -589,6 +1214,55 @@ json_unpack_auth_reply(struct pam_data *pd)
             }
             goto done;
         }
+
+        if (strncmp(key, "smartcard", strlen("smartcard")) == 0) {
+            ret = json_unpack_smartcard(tmp_ctx, jobj, &pin, &cai);
+            if (ret != EOK) {
+                goto done;
+            }
+
+            ret = sss_authtok_set_sc(pd->authtok, SSS_AUTHTOK_TYPE_SC_PIN,
+                                     pin, strlen(pin),
+                                     cai->token_name, strlen(cai->token_name),
+                                     cai->module_name, strlen(cai->module_name),
+                                     cai->key_id, strlen(cai->key_id),
+                                     cai->label, strlen(cai->label));
+            if (ret != EOK) {
+                DEBUG(SSSDBG_CRIT_FAILURE,
+                      "sss_authtok_set_sc failed: %d.\n", ret);
+            }
+            goto done;
+        }
+
+        if (strcmp(key, "passkey") == 0) {
+            ret = json_unpack_passkey(jobj, &pin, &passkey_kerberos, &crypto_challenge);
+            if (ret != EOK) {
+                goto done;
+            }
+
+            if (passkey_kerberos) {
+                if (pin != NULL && pin[0] != '\0') {
+                    user_verification = talloc_strdup(tmp_ctx, "true");
+                } else {
+                    user_verification = talloc_strdup(tmp_ctx, "false");
+                }
+                ret = sss_authtok_set_passkey_krb(pd->authtok, user_verification, crypto_challenge, pin);
+                if (ret != EOK) {
+                    DEBUG(SSSDBG_CRIT_FAILURE,
+                          "sss_authtok_set_passkey_krb failed: %d.\n", ret);
+                    goto done;
+                }
+            } else {
+                ret = sss_authtok_set_local_passkey_pin(pd->authtok, pin);
```

Comment:
I think `sss_authtok_set_passkey_krb()` and `sss_authtok_set_local_passkey_pin()` should be consistent in either requiring a `user_verification` argument or not.